ibm/resource_ibm_compute_vm_instance.go

```diff
@@ -157,6 +157,14 @@ func resourceIBMComputeVmInstance() *schema.Resource {
 				Computed: true,
 			},
 
+			"public_security_groups": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
```

Comment:
Can we have this exported as `public_security_group_ids`? Same with the private counterpart. 
Typically the security groups will be created using terraform itself and then its id can be easily interpolated and used here. It also matches with what the SL API expects and this case it is a good candidate to keep it that way.

In cases security groups are created outside terraform we can provide a datasource and use that. There is also a mechanism of importing existing ones.

Reply:
https://github.com/IBM-Bluemix/terraform-provider-ibm/pull/29/commits/468b4e658c5a2cbd823c9de166e745fb45dcacd0#diff-e4b220657319756663dd781aea77b4d0R446

will be called for each VM to translate the name to id, not an efficient way if someone spins 100 of VMs or so. 

It should be an ID. We will add a DS which gets us the ID of the group by taking name as input and use it anywhere, any number of times we like.